lms/djangoapps/teams/views.py

```diff
@@ -324,40 +340,49 @@ def get(self, request):
             result_filter.update({'topic_id': request.QUERY_PARAMS['topic_id']})
         if 'include_inactive' in request.QUERY_PARAMS and request.QUERY_PARAMS['include_inactive'].lower() == 'true':
             del result_filter['is_active']
-        if 'text_search' in request.QUERY_PARAMS:
-            return Response(
-                build_api_error(ugettext_noop("text_search is not yet supported.")),
-                status=status.HTTP_400_BAD_REQUEST
-            )
 
         queryset = CourseTeam.objects.filter(**result_filter)
         # We will always use name as either a primary or secondary sort.
         queryset = queryset.extra(select={'lower_name': "lower(name)"})
 
-        order_by_input = request.QUERY_PARAMS.get('order_by', 'name')
-        if order_by_input == 'name':
-            queryset = queryset.order_by('lower_name')
-        elif order_by_input == 'open_slots':
-            queryset = queryset.annotate(team_size=Count('users'))
-            queryset = queryset.order_by('team_size', 'lower_name')
-        elif order_by_input == 'last_activity':
-            return Response(
-                build_api_error(ugettext_noop("last_activity is not yet supported")),
-                status=status.HTTP_400_BAD_REQUEST
-            )
+        if 'text_search' in request.QUERY_PARAMS:
+            if CourseTeamIndexer.search_is_enabled():
+                search_engine = CourseTeamIndexer.engine()
+                text_search = request.QUERY_PARAMS['text_search'].encode('utf-8')
+
+                search_result = search_engine.search_string(text_search, size=MAXIMUM_SEARCH_SIZE)
+
+                results = search_result.get('results', [])
+                search_queryset_pks = [item['data']['pk'] for item in results]
+                queryset = queryset.filter(pk__in=search_queryset_pks)
+            else:
+                log.info(u"Text search requested but disabled")
```

Comment:
It looks like this is actually not handled server side, but only client side.  The search bar is enabled with a feature flag.  That same flag controls whether or not we need to initialize elastic search.

https://github.com/edx/edx-platform/blob/460a4b5f994b91eba1b696edcbd029d058f7f183/lms/envs/aws.py#L623
https://github.com/edx/edx-platform/blob/74c7c2ad77ff4af8c93ab14b5664eba905721ed8/lms/templates/dashboard.html#L131